lib/services.js

```diff
@@ -95,54 +96,36 @@ const services = {
         log.trace('storing object in metadata');
         assert.strictEqual(typeof bucketName, 'string');
-        const omVal = {};
-        // Version 2 changes the format of the location property
-        omVal['md-model-version'] = 2;
+        const md = new ObjectMD();
         // If an IAM user uploads a resource,
         // the owner should be the parent account.
         // http://docs.aws.amazon.com/AmazonS3/
         // latest/dev/access-control-overview.html
-        omVal['owner-display-name'] =
-            authInfo.getAccountDisplayName();
+        md.setOwnerDisplayName(authInfo.getAccountDisplayName());
         // This should be object creator's canonical ID.
-        omVal['owner-id'] = authInfo.getCanonicalID();
-        omVal['cache-control'] = cacheControl;
-        omVal['content-disposition'] = contentDisposition;
-        omVal['content-encoding'] = contentEncoding;
-        omVal.expires = expires;
-        omVal['content-length'] = size;
-        omVal['content-type'] = contentType;
+        md.setOwnerId(authInfo.getCanonicalID())
+            .setCacheControl(cacheControl)
+            .setContentDisposition(contentDisposition)
+            .setContentEncoding(contentEncoding)
+            .setExpires(expires)
+            .setContentLength(size)
+            .setContentType(contentType)
+            .setContentMd5(contentMD5)
+            .setLocation(dataGetInfo);
         // Sending in last modified date in object put copy since need
         // to return the exact date in the response
-        omVal['last-modified'] = lastModifiedDate || new Date().toJSON();
-        omVal['content-md5'] = contentMD5;
-
-        omVal['x-amz-server-version-id'] = '';
-        // TODO: Handle this as a utility function for all object puts
-        // similar to normalizing request but after checkAuth so
-        // string to sign is not impacted.  This is GH Issue#89.
-        omVal['x-amz-storage-class'] = 'STANDARD';
-        omVal['x-amz-server-side-encryption'] = '';
-        omVal['x-amz-server-side-encryption-aws-kms-key-id'] = '';
-        omVal['x-amz-server-side-encryption-customer-algorithm'] = '';
-        omVal.location = dataGetInfo;
-
+        if (lastModifiedDate) {
+            md.setLastModified(lastModifiedDate);
+        }
         if (cipherBundle) {
-            omVal['x-amz-server-side-encryption'] = cipherBundle.algorithm;
+            md.setAmzServerSideEncryption(cipherBundle.algorithm);
             if (cipherBundle.masterKeyId) {
-                omVal['x-amz-server-side-encryption-aws-kms-key-id'] =
-                    cipherBundle.masterKeyId;
+                md.setAmzEncryptionKeyId(cipherBundle.masterKeyId);
             }
         }
-
-        omVal.acl = {
-            Canned: 'private',
-            FULL_CONTROL: [],
-            WRITE_ACP: [],
-            READ: [],
-            READ_ACP: [],
-        };
-
+        if (headers && headers['x-amz-website-redirect-location']) {
```

Comment:
Saving the website redirect location seems to have been introduced in this commit, maybe deserve a special mention or separate commit (if not separate issue)?